### PR TITLE
Drop python 2 support

### PR DESCRIPTION
**Pull request by peymanslh**
**Issue:** [#390 ](https://github.com/jazzband/tablib/issues/390)

- Remove support python 2 from the docs, `requirements.txt`, `travis-ci`, `setup.py`, `tox.ini`
- Replace `unicode` with `str`
- Remove `dbfpy` folder and rename `dbfpy3` to `dbfpy`
- Remove `compat` file and remove python2 packages from dependency

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -28,3 +28,4 @@ Here is a list of passed and present much-appreciated contributors:
     Tommy Anthony
     Tsuyoshi Hombashi
     Tushar Makkar
+    Peyman Salehi
```
Comment (claudep) [inline AUTHORS]: Please keep the list alphabetically sorted.

File: tablib/core.py [hunk 2]
```diff
@@ -175,7 +173,7 @@ def __len__(self):
         return self.height
 
     def __getitem__(self, key):
-        if isinstance(key, (str, unicode)):
+        if isinstance(key, (str, str)):
```
Comment (claudep) [inline tablib/core.py]: Obviously one `str` too much :-) Same below.

File: tablib/formats/_dbf.py [hunk 3]
```diff
@@ -1,24 +1,13 @@
...
-    from tablib.packages.dbfpy3 import record as dbfrecord
-    import io
-else:
-    from tablib.packages.dbfpy import dbf
-    from tablib.packages.dbfpy import dbfnew
-    from tablib.packages.dbfpy import record as dbfrecord
+from tablib.packages.dbfpy import dbf
+from tablib.packages.dbfpy import dbfnew
+from tablib.packages.dbfpy import record as dbfrecord
+import io
```
Comment (claudep) [inline tablib/formats/_dbf.py]: I would move the `import io` at the top with other standard lib imports.

File: tablib/formats/_rst.py [hunk 4]
```diff
@@ -1,18 +1,9 @@
...
 
 from textwrap import TextWrapper
-
-from tablib.compat import (
-    median,
-    unicode,
-    izip_longest,
-)
+from statistics import median
+from itertools import zip_longest
```
Comment (claudep) [inline tablib/formats/_rst.py]: Please sort imports alphabetically.

**Comment by peymanslh** [comment]
Thanks for your review.
I fixed these issues and some other linting errors and updated my pr

File: setup.py [hunk 5]
```diff
@@ -63,14 +61,12 @@
         'Natural Language :: English',
         'License :: OSI Approved :: MIT License',
         'Programming Language :: Python',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
```
Comment (hugovk) [inline setup.py]: Can add this too:
```
'Programming Language :: Python :: 3 :: Only',
```

File: setup.py [hunk 6]
```diff
@@ -63,14 +61,12 @@
...
     ],
-    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*',
+    python_requires='!=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*',
```
Comment (hugovk) [inline setup.py]: We can simplify this:
```suggestion
    python_requires='>=3.4',
```

Comment (peymanslh) [reply setup.py]: Fixed

File: setup.py (hunk 5 above)
Comment (peymanslh) [reply setup.py]: Fixed

File: setup.py [hunk 7]
```diff
@@ -63,14 +60,13 @@
...
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3 :: Only',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
     ],
-    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*',
+    python_requires='>=3.4',
```
Comment (hugovk) [inline setup.py]: Sorry, I should have said:

```suggestion
    python_requires='>=3.5',
```

Comment (peymanslh) [reply setup.py]: Fixed :+1:

**Comment by claudep** [comment]
@peymanslh I'm sorry that recent commits broke your PR. Are you available to rebase it? Otherwise I can do it too.

**Comment by hugovk** [comment]
The big changes (from https://github.com/jazzband/tablib/pull/398) are to move `tablib` into a `src` directory, and `test_tablib.py` into a `tests` dir.

**Comment by peymanslh** [comment]
@claudep @hugovk 
Yes, I'm working on it.

**Comment by claudep** [comment]
Many thanks for your work, I just pushed the main part in master in b3c7145c407 (outside of this PR) and will finish by cherry-picking the linting fixes in a separate commit.

**Comment by claudep** [comment]
Done in bdc84255a83. Great work!

**Pull request closed without merging**